### PR TITLE
[stable10] Fix 500 when uploading in to share without write/edit permissions

### LINKED ISSUE #33416: Unexpected error code with chunk upload of files to folders with no edit permissions
### Steps to reproduce
1. **User A** shares a folder with **user B**, with no edit permissions
2. **User B** uploads a file to the shared folder

### Expected behaviour
After uploading all the file chunks, the `MOVE` operation responsible for moving the file to the final destination should retrieve a `403` error instead of `500` internal server error. 

We expect that the final `MOVE` tells us whether the operation can be done or not. 

### Actual behaviour
Final `MOVE` operation retrieves `500` internal error. 

### Server configuration

**Web server:** Apache/2.4.25 (Debian)

**Database:** MySQL

**PHP version:** php 7.2.11

**ownCloud version:** 10.0.10 stable

**Updated from an older ownCloud or fresh install:** fresh install

### Video with the step 2 described above

chunks_upload

### Related issues

https://github.com/owncloud/android/issues/2070#issuecomment-435377083
https://github.com/owncloud/android/issues/2291#issuecomment-434760233

### PR DESCRIPTION
**Pull request by IljaN**
## Description
Throw Forbidden exception if part-storage is not writable before write attempt.

## Related Issue

- Fixes #33416 

## How Has This Been Tested?
Acceptance test + manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by PVince81** [comment]
litmus not happy, is it an issue with the setup of the new test ?
```
3. put_get............... FAIL (PUT of `/remote.php/webdav/litmus/res' failed: 403 Forbidden)
4. put_get_utf8_segment.. 4. put_get_utf8_segment.. FAIL (PUT of `/remote.php/webdav/litmus/res-%e2%82%ac' failed: 403 Forbidden)
```

**Comment by IljaN** [comment]
Failed because of segfault (restarting)

**Comment by PVince81** [comment]
the test code looks different than https://github.com/owncloud/core/pull/33734

please sync it up then ping for review

**Comment by IljaN** [comment]
argh rebase fail

File: tests/acceptance/features/apiShareOperations/uploadToShare.feature [hunk 1]
```diff
@@ -297,3 +297,17 @@ Feature: sharing
     And parameter "shareapi_allow_public_upload" of app "core" has been set to "yes"
     When the public uploads file "test.txt" with content "test" using the public WebDAV API
     Then the content of file "/FOLDER/test.txt" for user "user0" should be "test"
+
+  Scenario: Uploading a file in to a shared folder without edit permissions
+    Given using new DAV path
+    And user "user1" has been created with default attributes
+    And user "user0" creates a folder "/READ_ONLY" using the WebDAV API
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareOperations/uploadToShare.feature]: Needs this test step refactored. See #33884 
```
And user "user0" creates folder "/READ_ONLY" using the WebDAV API
```

Comment (phil-davis) [reply tests/acceptance/features/apiShareOperations/uploadToShare.feature]: I rebased and cherry-picked the commit to fix this

**Comment by phil-davis** [comment]
I rebased and cherry-picked the acceptance test fix from #33884

**Comment by IljaN** [comment]
good timing, just did the same ;)

**Comment by phil-davis** [comment]
All good :+1:

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**